suite/src/cmd.rs

```diff
@@ -0,0 +1,38 @@
+use alloc::{format, vec::Vec};
+use benchmark_common::{BenchmarkResult, IncomingMessage, OutgoingMessage, SuiteStatus};
+
+use crate::{
+    benchmark::examples,
+    platform::{self, Platform},
+};
+
+/// Takes an IncomingMessage and decides how to respond to it
+///
+/// # Arguments
+///
+/// * `cmd` - the message to produce a response to
+pub fn run_cmd(cmd: IncomingMessage) -> Option<OutgoingMessage> {
+    match cmd {
+        IncomingMessage::Suspend(value) => platform::current().suspend(value),
+        IncomingMessage::Invalid(msg) => {
+            Some(OutgoingMessage::Error(format!("Invalid message: {msg}")))
+        }
+        IncomingMessage::Done => Some(OutgoingMessage::Status(SuiteStatus::Done)),
+        IncomingMessage::GetStatus => Some(OutgoingMessage::Status(SuiteStatus::Ready)),
+        IncomingMessage::Benchmark(info, n) => {
+            let mut results: Vec<BenchmarkResult> = Vec::new();
+
+            for _ in 0..n {
+                let result = match info {
```

Comment:
Is this match exhaustive? Maybe you are missing a "_" identifier as the second match case?

Reply:
Thanks to rust, this code would not compile if the match was not exhaustive.
There will be different benchmarks added in the future for all the other modules but for now there is only this one.

I intentionally did not use any '_' or other catch-all patterns, because if I modify the messages that can be sent in the common crate, the linter/compiler immediately shows me every place in the suite & cli where I have to now consider this new message types. 
This is one of the main reasons for the common crate, preventing me from forgetting to correctly handle a valid message.

Reply:
Ok, makes sense. I was just worried that it was an overkill if the match had only one arm.